contrib/platform/src/com/sun/jna/platform/win32/Psapi.java

```diff
@@ -320,4 +303,85 @@ class PERFORMANCE_INFORMATION extends Structure {
         public DWORD ProcessCount;
         public DWORD ThreadCount;
     }
+
+    @FieldOrder({"Flags"})
+    class PSAPI_WORKING_SET_EX_BLOCK extends Structure {
+        public ULONG_PTR[] Flags = new ULONG_PTR[Native.POINTER_SIZE == 8 ? 1 : 2];
+        private long innerValue;
+
+        @Override
+        public void read() {
+            super.read();
+            if (innerValue == 0l) {
+                return;
+            } else if (this.Flags[0] == null) {
+                return;
+            } else {
+                innerValue = this.Flags[0].longValue();
+            }
+        }
+
+        public boolean Valid() {
+            return getBitFieldValue(1) == 1;
+        }
+
+        public int ShareCount() {
+            return getBitFieldValue(3, 1);
+        }
+
+        public int Wind32Protection() {
+            return getBitFieldValue(11, 3, 1);
+        }
+
+        public boolean Shared() {
+            return getBitFieldValue(1, 11, 3, 1) == 1;
+        }
+
+        public int Node() {
+            return getBitFieldValue(6, 1, 11, 3, 1);
+        }
+
+        public boolean Locked() {
+            return getBitFieldValue(1, 6, 1, 11, 3, 1) == 1;
+        }
+
+        public boolean LargePage() {
+            return getBitFieldValue(1, 1, 6, 1, 11, 3, 1) == 1;
+        }
+
+        public boolean Reserved() {
+            return getBitFieldValue(1, 1, 1, 6, 1, 11, 3, 1) == 1;
+        }
+
+        public boolean Bad() {
+            return getBitFieldValue(1,1, 1, 1, 6, 1, 11, 3, 1) == 1;
+        }
+
+        public long ReservedUlong() {
+            return innerValue >> 32;
+        }
+
+        private int getBitFieldValue(int intLength, int... otherBits) {
```

Comment:
I've been trying to find more uses for VarArgs recently. For stuff like this I like it, since I'm assuming the JVM will eventually realize that it's a constant value for some functions and optimize it for me.

Reply:
If you want compiler optimization just put a + sign in between the other bits instead of a comma. :)